axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/LegacyAxonServerEventStorageEngine.java

```diff
@@ -194,7 +195,11 @@ public MessageStream<EventMessage<?>> source(@Nonnull SourcingCondition conditio
     }
 
     private MessageStream<EventMessage<?>> eventsForCriteria(SourcingCondition condition, EventCriteria criterion) {
-        String aggregateIdentifier = resolveAggregateIdentifier(criterion.tags());
+        if (!(criterion instanceof TagEventCriteria)) {
```

Comment:
Do we want any TagEventCriteria or maybe specialize it to some type like `AggregateEventCriteria`? Because in legacy we don't have tags, just AggregateId.